indexer-standalone/src/main.rs

```diff
@@ -146,31 +147,39 @@ async fn run() -> anyhow::Result<()> {
         )
     });
 
+    info!("indexer shutting down");
+
     select! {
         result = chain_indexer => handle_exit("chain-indexer", result),
         result = wallet_indexer => handle_exit("wallet-indexer", result),
         result = indexer_api => handle_exit("indexer-api", result),
     }
 
-    info!("indexer shutting down");
-
     Ok(())
 }
 
 #[cfg(feature = "standalone")]
-fn handle_exit(task_name: &str, result: Result<anyhow::Result<()>, tokio::task::JoinError>) {
+fn handle_exit(task_name: &str, result: Result<anyhow::Result<()>, tokio::task::JoinError>) -> ! {
     use log::error;
+    use std::process;
 
     match result {
         Ok(Err(error)) => {
             let backtrace = error.backtrace();
             let error = format!("{error:#}");
-            error!(error, backtrace:%; "{task_name} exited with ERROR")
+            error!(error, backtrace:%; "{task_name} exited with ERROR");
+            process::exit(1);
         }
 
-        Err(error) => error!(error:% = format!("{error:#}"); "{task_name} panicked"),
+        Err(error) => {
+            error!(error:% = format!("{error:#}"); "{task_name} panicked");
+            process::exit(1);
+        }
 
-        _ => error!("{task_name} terminated"),
+        _ => {
+            error!("{task_name} exited successfully");
```

Comment:
Now this is a tricky special case: None of the components can exit successfully (if it could, it were a bug). Hence each of them already logs something like "block_indexed_task completed unexpectedly". So what should we do here? I think the best way would be to simply replace "successfully" with "unexpectedly" AND then exit with `1` instead of `0`.